src/include/denc.h

```diff
@@ -94,7 +94,7 @@ inline constexpr bool denc_supported = denc_traits<T>::supported;
     if (bits > 2)							\
       break;								\
     char fn[PATH_MAX];							\
-    snprintf(fn, sizeof(fn),						\
+    ::snprintf(fn, sizeof(fn),						\
```

Comment:
@iotcg could you share with us what build error you are facing without this change? i cannot find another declaration/implementation of `snprintf()`  "in local namespace" .

Reply:
**file: src/include/utime.h**
    1 +-- 28 lines: -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*- -----
   29 #include "include/**denc.h**"
   30 +-- 13 lines: -------------------------------------------------------------------------
   43 class utime_t {
   44 +--108 lines: public:------------------------------------------------------------------
  152   **DENC**(utime_t, v, p) {
  153     denc(v.tv.tv_sec, p);
  154     denc(v.tv.tv_nsec, p);
  155   }
  156 +--182 lines: void encode_timeval(struct ceph_timespec *t) const {----------------------
  338   **static int snprintf**(char *out, int outlen, time_t tt) {
  339     struct tm bdt;
  340     localtime_r(&tt, &bdt);
  341 
  342     return ::snprintf(out, outlen,
  343         "%04d-%02d-%02d %02d:%02d:%02d",
  344         bdt.tm_year + 1900, bdt.tm_mon + 1, bdt.tm_mday,
  345         bdt.tm_hour, bdt.tm_min, bdt.tm_sec);
  346   }
  347 
  348 +---100 lines: static int invoke_date(const std::string& date_str, utime_t *result) {----
  448 };
  449 +-- 93 lines: WRITE_CLASS_ENCODER(utime_t)------------------------------------------------
  
  
**file: src/include/denc.h**  
    1 +-- 86 lines: -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*---------
   87 # define **DENC_DUMP_POST**(Type)                   \
   88 +--  9 lines: do {         \------------------------------------------------------------
   97     **snprintf**(fn, sizeof(fn),                                            \
   98              ENCODE_STRINGIFY(ENCODE_DUMP_PATH) "/%s__%d.%x", #Type,            \
   99              getpid(), i++);                                            \
  100 +--1588 lines: int fd = ::open(fn, O_WRONLY|O_TRUNC|O_CREAT|O_CLOEXEC, 0644);  \--------
 1688 #define **DENC**(Type, v, p)                                                \
 1689   DENC_HELPERS                                                          \
 1690   void bound_encode(size_t& p) const {                                  \
 1691     _denc_friend(*this, p);                                             \
 1692   }                                                                     \
 1693   void encode(bufferlist::contiguous_appender& p) const {               \
 1694     DENC_DUMP_PRE(Type);                                                \
 1695     _denc_friend(*this, p);                                             \
 1696     **DENC_DUMP_POST(**Type);                                               \
 1697   }                                                                     \
 1698   void decode(buffer::ptr::const_iterator& p) {                         \
 1699     _denc_friend(*this, p);                                             \
 1700   }                                                                     \
 1701   template<typename T, typename P>                                      \
 1702   friend std::enable_if_t<std::is_same_v<T, Type> ||                    \
 1703                           std::is_same_v<T, const Type>>                \
 1704   _denc_friend(T& v, P& p)
 1705 +-- 20 lines: #define DENC_FEATURED(Type, v, p, f)     \-----------------

Reply:
@tchaikov 
1. Currently, I don't have the build error log.
2. The above code shows: If global namespace snprintf is not used, DENC_DUMP_POST will call utime_t::snprintf function. However, their function protertype doesn't match.